File: src/main/java/mountainrangepvp/net/ClientInterface.java
package mountainrangepvp.net;

/**
 * The interface the server talks to, either in-process or over the network.
 */
public interface ClientInterface {
    /**
     * Tells a client that they've successfully connected.
     */
    public void connected(ClientId id);

    public void sessionInfo(boolean teamsOn);

    public void newMap(int seed);
}

File: src/main/scala/mountainrangepvp/net/ClientInterface.scala
package mountainrangepvp.net

/**
 * The interface the server talks to, either in-process or over the network.
 */
trait ClientInterface {
  /**
   * The server closes the connection.
   */
  def disconnected()

  /**
   * Receive a message from the server.
   */
  def deliver(message: ToClientMessage)
}
